packages/legend-extension-application-studio-query-builder/src/components/ServiceQueryBuilder.tsx

```diff
@@ -69,6 +69,11 @@ export const ServiceQueryBuilder = observer(
             queryBuilderExtension.queryBuilderState.changeDetectionState.setQueryHashCode(
               hashObject(executionState.execution.func),
             );
+            await Promise.all([
```

Comment:
See my comment in `QueryBuilderChangeDetectionState`, we should only need to do:
`await queryBuilderState.changeDetectionState.start();` here